autosklearn/automl.py

```diff
@@ -756,10 +769,12 @@ def score(self, X, y):
 
         prediction = self.predict(X)
 
-        # Prediction on classification needs to be decoded for score
-        # That is, when doing bool/categorical predictions, a categorical
-        # prediction is returned by the estimator, yet we compare a
-        # numerical metric
+        # Encode the prediction using the input validator
+        # We train autosklearn with a encoded version of y,
+        # which is decoded by predict().
+        # Above call to validate() encodes the y given for score()
+        # Below call encodes the prediction, so we compare in the
+        # same representation domain
```

Comment:
Looking at this once more, don't we have probabilities here? How can you encode here?

Reply:
During prediction, we take the indices in here:
https://github.com/automl/auto-sklearn/blob/54c8f9781f163c6316b6f6d0ed150f39d524c0e9/autosklearn/automl.py#L1010

So I don't think you get probabilities. If the user wants to get them, he can use predict_proba. Let me know if I am missing a use case!

Reply:
Okay, I now understand what's going on.